### PR TITLE
VideoCore: Inline IsPicaTracing

### PR DESCRIPTION
**Pull request by yuriks**
Speeds up ALBW main menu slightly (~3%)

### CODE REVIEW

File: src/video_core/debug_utils/debug_utils.cpp [hunk 1]
```diff
@@ -276,45 +276,37 @@ void DumpShader(const std::string& filename, const Regs::ShaderConfig& config,
...
-    return is_pica_tracing != 0;
+    g_is_pica_tracing = true;
 }
 
 void OnPicaRegWrite(PicaTrace::Write write) {
-    // Double check for is_pica_tracing to avoid pointless locking overhead
-    if (!is_pica_tracing)
-        return;
-
     std::lock_guard<std::mutex> lock(pica_trace_mutex);
```
Comment (JayFoxRox) [inline src/video_core/debug_utils/debug_utils.cpp]: scope around this one so it only gets created when tracing is activated?

Comment (JayFoxRox) [inline src/video_core/debug_utils/debug_utils.cpp]: nvm, just noticed that there is only one user now and the following if check will always be true..
Makes me wonder if we should just seperate `is_pica_tracing` from this fuction entirely, then document that caller is responsible for only calling it during tracing.

Comment (yuriks) [inline src/video_core/debug_utils/debug_utils.cpp]: I'm not sure what you mean? (This only gets called when tracing is active in the first place.)

Comment (yuriks) [inline src/video_core/debug_utils/debug_utils.cpp]: The second check is there to ensure that the value is correct while the lock is held. It's a crappy implementation of double-checked locking.

**Comment by JayFoxRox** [comment]
LGTM

**Pull request merged**